### PR TITLE
[release-4.10] Bug 2100894: fix MCNameSuffix with kcfg ctrrcfg

### PR DESCRIPTION
**Pull request by QiWang19**
Fix: https://bugzilla.redhat.com/show_bug.cgi?id=2100894
Backport https://github.com/openshift/machine-config-operator/pull/2936 and https://github.com/openshift/machine-config-operator/pull/3093. This PR will fix the issue that the latest kubelet config and containerruntime config get overridden by the first one. Fix also works if the first config came from the bootstrap mode.

**- What I did**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Review by yuqi-zhang — APPROVED** [review]
Did a general look-over, the cherry pick looks good to me

**Comment by yuqi-zhang** [comment]
/label backport-risk assessed

Should be safe behaviour fixes for relatively edge caess

File: pkg/controller/container-runtime-config/container_runtime_config_bootstrap_test.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestAddContainerRuntimeCfgAfterBootstrapContainerRuntimeCfg(t *testing.T) {
+	for _, platform := range []apicfgv1.PlatformType{apicfgv1.AWSPlatformType, apicfgv1.NonePlatformType, "unrecognized"} {
+		t.Run(string(platform), func(t *testing.T) {
+			f := newFixture(t)
+			f.newController()
+
+			cc := newControllerConfig(ctrlcommon.ControllerConfigName, platform)
```
[the remark this answers is not in the document]
Comment (QiWang19) [reply pkg/controller/container-runtime-config/container_runtime_config_bootstrap_test.go:20]: This test the case explicitly.

**Comment by yuqi-zhang** [comment]
/label backport-risk-assessed

typo previously

**Pull request merged**